README.md

```diff
@@ -243,6 +242,56 @@ In the strict mode, `async` functions are checked such that they only raise
 effects on forward declarations, callbacks and methods using
 `{.raises: [CatchableError].}` (or more strict) annotations.
 
+### Cancellation support
+
+Any running `Future` can be cancelled.
+
+```nim
+proc c1 {.async.} =
+  echo "Before sleep"
+  try:
+    await sleepAsync(10.minutes)
+    echo "After sleep" # never reached
+  except CancelledError as exc:
+    echo "We got cancelled!"
+    raise exc
+
+proc c2 {.async.} =
+  await c1()
+  echo "Never reached, since the CancelledError bubble up"
+
+let work = c2()
+waitFor(work.cancelAndWait())
+```
+
+The cancellation will travel down the call stack, and throw a `CancelledError`
+on the currently running `await` (in this case the `await sleepAsync` in `c1`)
+
+The `CancelledError` will now travel up the stack like any other exception.
+It can be caught and handled (for instance, freeing some resources), or it can
+swallowed if a task must not be cancelled.
+
+Since `CancelledError` inherits from `CatchableError`, be careful to handle it
+appropriately:
+
+```nim
+proc c3 {.async.} =
+  try:
+    await somethingThatCanFail()
+  except CatchableError as exc:
+    echo "Something went wrong: ", exc.msg
+
+proc c4 {.async.} =
+  while true:
+    await c3()
+
+waitFor(c4().wait(10.seconds))
+```
+
+In this example, `c3` swallows cancellations, so `c4` cannot be cancelled
+properly, and the `wait` timeout will not work
```

Comment:
'с3' swallows cancellations, but why `wait` timeout will not work? This code will exit from `waitFor` but because cancellation was swallowed, `c4` Future will never be completed.

Reply:
```nim
import chronos
proc c3 {.async.} =
  try:
    await sleepAsync(100.minutes)
  except CatchableError as exc:
    echo "Something went wrong: ", exc.msg

proc c4 {.async.} =
  while true:
    await c3()

waitFor(c4().wait(5.seconds))
echo "Never reached"
```